Sources/Widgets/Widgets3D/ResliceCursorWidget/helpers.js

```diff
@@ -0,0 +1,223 @@
+import vtkBoundingBox from 'vtk.js/Sources/Common/DataModel/BoundingBox';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+
+import { ViewTypes } from 'vtk.js/Sources/Widgets/Core/WidgetManager/Constants';
+
+// Return the closest point from the pointsList to the refPoint
+export function getClosestPoint(refPoint, points) {
+  let minDist = Number.MAX_VALUE;
+  let minPoint = null;
+  points.forEach((point) => {
+    const dist = vtkMath.distance2BetweenPoints(refPoint, point);
+    if (dist < minDist) {
+      minPoint = point;
+      minDist = dist;
+    }
+  });
+  return minPoint;
+}
+
+// Project point (inPoint) to the bounds of the image according to a plane
+// defined by two vectors (v1, v2)
+export function boundPoint(inPoint, v1, v2, image) {
+  const outPoint = [];
+  const bounds = image.getBounds();
+
+  const absT1 = v1.map((val) => Math.abs(val));
+  const absT2 = v2.map((val) => Math.abs(val));
+  const epsilon = 0.00001;
+
+  let o1 = 0.0;
+  let o2 = 0.0;
+
+  for (let i = 0; i < 3; i++) {
+    let axisOffset = 0;
+
+    const useT1 = absT1[i] > absT2[i];
+    const t = useT1 ? v1 : v2;
+    const absT = useT1 ? absT1 : absT2;
+
+    if (inPoint[i] < bounds[i * 2]) {
+      axisOffset = absT[i] > epsilon ? (bounds[2 * i] - inPoint[i]) / t[i] : 0;
+    } else if (inPoint[i] > bounds[2 * i + 1]) {
+      axisOffset =
+        absT[i] !== epsilon ? (bounds[2 * i + 1] - inPoint[i]) / t[i] : 0;
+    }
+
+    if (useT1) {
+      if (Math.abs(axisOffset) > Math.abs(o1)) {
+        o1 = axisOffset;
+      }
+    } else if (Math.abs(axisOffset) > Math.abs(o2)) {
+      o2 = axisOffset;
+    }
+  }
+
+  outPoint[0] = inPoint[0];
+  outPoint[1] = inPoint[1];
+  outPoint[2] = inPoint[2];
+
+  if (o1 !== 0.0) {
+    const translation = [];
+
+    translation[0] = v1[0] * o1;
+    translation[1] = v1[1] * o1;
+    translation[2] = v1[2] * o1;
+
+    vtkMath.add(outPoint, translation, outPoint);
+  }
+  if (o2 !== 0.0) {
+    const translation = [v2[0] * o2, v2[1] * o2, v2[2] * o2];
+    vtkMath.add(outPoint, translation, outPoint);
+  }
+
+  return outPoint;
+}
+
+// Get name of the line in the same plane as the input
+export function getAssociatedLinesName(lineName) {
+  switch (lineName) {
+    case 'axisXinY':
+      return 'AxisZinY';
+    case 'axisXinZ':
+      return 'AxisYinZ';
+    case 'axisYinX':
+      return 'AxisZinX';
+    case 'axisYinZ':
+      return 'AxisXinZ';
+    case 'axisZinX':
+      return 'AxisYinX';
+    case 'axisZinY':
+      return 'AxisXinY';
```

Comment:
the casing inconsistency is puzzling

Reply:
I updated it with uppercase. Because the generated getter need uppercase for the first letter.

Reply:
Ok, but you could have used the `capitalize()` function in macro.js

Reply:
I didn't know this function...

Reply:
The getters use internally the `capitalize()` function to generate the proper `getFieldName()` method when you provide `macro.get(publicAPI, model, ['fieldName'])`.